sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -261,22 +271,20 @@ Status getStatus() {
    */
   @VisibleForTesting
   List<Link> getLinks() {
-    synchronized (lock) {
-      if (links == null) {
-        return Collections.emptyList();
-      }
-      List<Link> result = new ArrayList<>(links.size());
-      for (Link link : links) {
-        Link newLink = link;
-        if (!(link instanceof SpanData.Link)) {
-          // Make a copy because the given Link may not be immutable and we may reference a lot of
-          // memory.
-          newLink = SpanData.Link.create(link.getContext(), link.getAttributes());
-        }
-        result.add(newLink);
+    if (links == null) {
+      return Collections.emptyList();
+    }
+    List<Link> result = new ArrayList<>(links.size());
+    for (Link link : links) {
+      Link newLink = link;
+      if (!(link instanceof SpanData.Link)) {
+        // Make a copy because the given Link may not be immutable and we may reference a lot of
+        // memory.
+        newLink = SpanData.Link.create(link.getContext(), link.getAttributes());
       }
-      return Collections.unmodifiableList(result);
+      result.add(newLink);
     }
+    return Collections.unmodifiableList(result);
```

Comment:
Do you need to wrap this?

Reply:
We don't but the lint complains:
` warning: [MixedMutabilityReturnType] This method returns both mutable and immutable collections or maps from different paths. This may be confusing for users of the method.`

So I'd rather leave it as-is.